apps/accounts/views.py

```diff
@@ -72,29 +99,59 @@ def login_user(request, user, identity):
     login(request, user)
 
 
+def render_error(request, error, next_path=None):
+    if resolve(request.path).func.func_name == 'AuthenticateView':
```

Comment:
Yeah, that was my original thought but this was easier so I did it. Definitely felt wrong at the time and I kinda forgot about it. I was thinking:

```py
class AuthenticateView(APIView):
    @with_user(format='html')
    def get(self, ...):
        ...

class LoginView(APIView):
    @with_user(format='json')
    def post(self, ...):
        ...
```

It would probably be best to pull it from the `Accept` header. I guess giving JSON preference in the `*/*` scenario. I don't think we were telling anyone to specify that they want JSON in the response so that might be backwards incompatible (not that this specific API is used...).